### ISSUE TITLE
Avoid synchronous device ops for Gradient Accumulation API

### ISSUE THREAD

**Issue by rpsilva-aws** [pytorch/xla#9143]
## 🚀 Feature
Currently, we require syncing all inputs to the body function of the gradient accumulation (ref0), since these need to be device data nodes during the body computation staging. However, one of the arguments are iterable tensors, and these need to be sliced prior to staging the body function (ref1). The actual dynamic slice operation sits above the body function on the HLO, and we need to trace this function in advance (ref2).

Since the sliced iterable tensors are effectively dummy input tensors that need to be synced and therefore, require a synchronous operation, we ought replace them with placeholder tensors. This would allow us to have full asynchronous execution for the gradient accumulation APIs.

The problem is that placeholder tensors are resulting in a seg fault when used to stage the user computation: https://github.com/pytorch/xla/issues/9049
